rotkehlchen/tests/api/test_exchanges.py

```diff
@@ -1245,3 +1245,142 @@ def test_binance_query_pairs(rotkehlchen_api_server_with_exchanges):
     assert 'FTTBNB' not in result
     if ci_run is False:
         assert binance_pairs_num > binanceus_pairs_num
+
+
+@pytest.mark.parametrize('default_mock_price_value', [FVal('5.5')])
+@pytest.mark.parametrize('added_exchanges', [(Location.BINANCE,)])
+def test_get_binance_savings_balance(rotkehlchen_api_server_with_exchanges):
+    """Check that querying the binance savings balance endpoint returns the expected response."""
+    async_query = random.choice([True, False])
+
+    # check for errors
+    response = requests.get(
+        api_url_for(
+            rotkehlchen_api_server_with_exchanges,
+            'binancesavingsresource',
+            location=Location.KRAKEN.name.lower(),
+        ),
+    )
+    assert_error_response(response, 'is not one of binance,binanceus')
+
+    lending_balances_call_count = 0
+    interest_daily_call_count = 0
+    interest_customized_fixed_call_count = 0
+    interest_activity_call_count = 0
+
+    def mock_api_query(api_type, method, options):  # pylint: disable=unused-argument
+        nonlocal interest_daily_call_count, \
+            interest_customized_fixed_call_count, \
```

Comment:
since this is big maybe you can move to an auxiliar function in a utils file? It makes harder to read the test